lib/driver.js

```diff
@@ -980,7 +1007,7 @@ class XCUITestDriver extends BaseDriver {
           this.log.debug(`Unable to DELETE session on WDA: '${err.message}'. Continuing shutdown.`);
         }
       }
-      if (!this.wda.webDriverAgentUrl && this.opts.useNewWDA) {
+      if ((!this.wda.webDriverAgentUrl && this.opts.useNewWDA) || this.opts.usePreinstalledWDA) {
```

Comment:
note:
replace `this.opts.usePreinstalledWDA` with `this.wda.canSkipXcodebuild` if https://github.com/appium/WebDriverAgent/pull/691 is faster than this pr

Reply:
where do we actually perform capabilities validation (e.g. that the target device is not a simulator)?

Reply:
Perhaps `const usePreinstalledWDA = this.isRealDevice() && this.opts.usePreinstalledWDA;` in L751 is better. I'll add it

Reply:
added https://github.com/appium/appium-xcuitest-driver/pull/1609/files#diff-ed3c013edbc896bd87ef62539626fcdc8930a31f3cee7ecb37438e172af2c8e0R751